### PR TITLE
Document configurations

### PR DESCRIPTION
**Pull request by GalRogozinski**
# Description

Document configurations

## Type of change

- Documentation Fix

### CODE REVIEW

File: src/main/java/com/iota/iri/conf/MilestoneConfig.java [hunk 1]
```diff
@@ -6,12 +6,16 @@
...
-     * @return Descriptions#COORDINATOR
+     * Default Value: {@value BaseIotaConfig.Defaults#COORDINATOR_ADDRESS}
+     *
+     * @return {@value MilestoneConfig.Descriptions#COORDINATOR}
      */
     String getCoordinator();
 
     /**
-     * @return {@value Descriptions#DONT_VALIDATE_TESTNET_MILESTONE_SIG}
+     * Default Value: <tt>false</tt>
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/MilestoneConfig.java]: why break the pattern of `Default Value: {@value ...`?

Comment (alon-e) [reply src/main/java/com/iota/iri/conf/MilestoneConfig.java]: has a default: `DONT_VALIDATE_MILESTONE_SIG` - which doesn't match the description name :\

File: src/main/java/com/iota/iri/conf/SnapshotConfig.java [hunk 2]
```diff
@@ -6,67 +6,93 @@
...
-     * @return {@value Descriptions#LOCAL_SNAPSHOTS_BASE_PATH}
+     * Default Value: {@value BaseIotaConfig.Defaults#LOCAL_SNAPSHOTS_BASE_PATH}
+     *
+     * @return {@value SnapshotConfig.Descriptions#LOCAL_SNAPSHOTS_BASE_PATH}
      */
     String getLocalSnapshotsBasePath();
 
     /**
-     * @return {@value Descriptions#NUMBER_OF_KEYS_IN_A_MILESTONE}
+     * Default Value: {@value BaseIotaConfig.Defaults#NUM_KEYS_IN_MILESTONE}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/SnapshotConfig.java]: the default value name does not match the description name.
`NUMBER_OF_KEYS_IN_A_MILESTONE` vs `NUM_KEYS_IN_MILESTONE`.
it's confusing to know which of them to use in the INI file.

File: src/main/java/com/iota/iri/conf/SnapshotConfig.java [hunk 3]
```diff
@@ -6,67 +6,93 @@
...
+     *
+     * @return {@value SnapshotConfig.Descriptions#NUMBER_OF_KEYS_IN_A_MILESTONE}
      */
     int getNumberOfKeysInMilestone();
 
     /**
-     * @return {@value Descriptions#PREVIOUS_EPOCH_SPENT_ADDRESSES_FILE}
+     * Default Value: {@value BaseIotaConfig.Defaults#PREVIOUS_EPOCHS_SPENT_ADDRESSES_TXT}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/SnapshotConfig.java]: the default value name does not match the description name.
txt vs file

File: src/main/java/com/iota/iri/conf/TipSelConfig.java [hunk 4]
```diff
@@ -8,17 +8,23 @@
...
+    /**
+     * Default Value: {@value BaseIotaConfig.Defaults#ALPHA}
+     *
+     * @return {@value TipSelConfig.Descriptions#ALPHA}
      */
     double getAlpha();
 
     /**
-     * @return Descriptions#BELOW_MAX_DEPTH_TRANSACTION_LIMIT
+     * Default Value: {@value BaseIotaConfig.Defaults#MAX_ANALYZED_TXS}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/TipSelConfig.java]: the default value name does not match the description name.
`MAX_ANALYZED_TXS` vs `BELOW_MAX_DEPTH_TRANSACTION_LIMIT`

File: src/main/java/com/iota/iri/conf/SnapshotConfig.java [hunk 5]
```diff
@@ -6,67 +6,93 @@
...
-     * return {@value Descriptions#SNAPSHOT_FILE}
+     * Default Value: {@value BaseIotaConfig.Defaults#SNAPSHOT_FILE}
+     *
+     * return {@value SnapshotConfig.Descriptions#SNAPSHOT_FILE}
      */
     String getSnapshotFile();
 
     /**
-     * @return {@value Descriptions#SNAPSHOT_SIGNATURE_FILE}
+     * Default Value: {@value BaseIotaConfig.Defaults#SNAPSHOT_SIG_FILE}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/SnapshotConfig.java]: the default value name does not match the description name.
`SNAPSHOT_SIG_FILE` vs `SNAPSHOT_SIGNATURE_FILE`

File: src/main/java/com/iota/iri/conf/SnapshotConfig.java [hunk 6]
```diff
@@ -6,67 +6,93 @@
...
-     * @return {@value Descriptions#LOCAL_SNAPSHOTS_DEPTH}
+     * Default Value: {@value BaseIotaConfig.Defaults#LOCAL_SNAPSHOTS_DEPTH}
+     *
+     * @return {@value SnapshotConfig.Descriptions#LOCAL_SNAPSHOTS_DEPTH}
      */
     int getLocalSnapshotsDepth();
 
     /**
-     * @return {@value Descriptions#SNAPSHOT_TIME}
+     * Default Value: {@value BaseIotaConfig.Defaults#GLOBAL_SNAPSHOT_TIME}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/SnapshotConfig.java]: the default value name does not match the description name.
`GLOBAL_SNAPSHOT_TIME` vs `SNAPSHOT_TIME`

File: src/main/java/com/iota/iri/conf/ProtocolConfig.java [hunk 7]
```diff
@@ -7,34 +7,59 @@
...
+    /**
+     * Default Value: {@value BaseIotaConfig.Defaults#P_DROP_TRANSACTION}
+     *
+     * @return {@value ProtocolConfig.Descriptions#P_DROP_TRANSACTION}
+     */
     double getpDropTransaction();
 
     /**
-     * @return Descriptions#P_SELECT_MILESTONE
+     * Default Value: {@value BaseIotaConfig.Defaults#P_SELECT_MILESTONE_CHILD}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/ProtocolConfig.java]: the default value name does not match the description name.
`P_SELECT_MILESTONE_CHILD` vs `P_SELECT_MILESTONE`

File: src/main/java/com/iota/iri/conf/ProtocolConfig.java [hunk 8]
```diff
@@ -7,34 +7,59 @@
...
-     * @return Descriptions#TRANSACTION_PACKET_SIZE
+     * Default Value: {@value BaseIotaConfig.Defaults#PACKET_SIZE}
+     *
+     * @return {@value ProtocolConfig.Descriptions#TRANSACTION_PACKET_SIZE}
      */
     int getTransactionPacketSize();
 
     /**
-     * @return Descriptions#REQUEST_HASH_SIZE
+     * Default Value: {@value BaseIotaConfig.Defaults#REQ_HASH_SIZE}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/ProtocolConfig.java]: the default value name does not match the description name.
`REQ_HASH_SIZE` vs `REQUEST_HASH_SIZE`

File: src/main/java/com/iota/iri/conf/ProtocolConfig.java [hunk 9]
```diff
@@ -7,34 +7,59 @@
...
-     * @return Descriptions#MWM
+     * Default Value: {@value BaseIotaConfig.Defaults#MWM}
+     *
+     * @return {@value ProtocolConfig.Descriptions#MWM}
      */
     int getMwm();
 
     /**
-     * @return Descriptions#TRANSACTION_PACKET_SIZE
+     * Default Value: {@value BaseIotaConfig.Defaults#PACKET_SIZE}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/ProtocolConfig.java]: the default value name does not match the description name.
`PACKET_SIZE` vs `TRANSACTION_PACKET_SIZE`

File: src/main/java/com/iota/iri/conf/NetworkConfig.java [hunk 10]
```diff
@@ -9,57 +9,78 @@
...
-     * @return Descriptions#NEIGHBORS
+     * Default Value: {@code {}}
+     *
+     * @return {@value NetworkConfig.Descriptions#NEIGHBORS}
      */
     List<String> getNeighbors();
 
     /**
-     * @return Descriptions#Q_SIZE_NODE
+     * Default Value: {@value BaseIotaConfig.Defaults#QUEUE_SIZE}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/NetworkConfig.java]: the default value name does not match the description name.
`QUEUE_SIZE` vs `Q_SIZE_NODE`

File: src/main/java/com/iota/iri/conf/MilestoneConfig.java [hunk 11]
```diff
@@ -6,12 +6,16 @@
 public interface MilestoneConfig extends Config {
 
     /**
-     * @return Descriptions#COORDINATOR
+     * Default Value: {@value BaseIotaConfig.Defaults#COORDINATOR_ADDRESS}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/MilestoneConfig.java]: the default value name does not match the description name.
`COORDINATOR_ADDRESS` vs `COORDINATOR`

File: src/main/java/com/iota/iri/conf/DbConfig.java [hunk 12]
```diff
@@ -6,32 +6,44 @@
...
-     * @return {@value Descriptions#DB_CACHE_SIZE}
+     * Default Value: {@value BaseIotaConfig.Defaults#DB_CACHE_SIZE}
+     *
+     * @return {@value DbConfig.Descriptions#DB_CACHE_SIZE}
      */
     int getDbCacheSize();
 
     /**
-     * @return {@value Descriptions#MAIN_DB}
+     * Default Value: {@value BaseIotaConfig.Defaults#ROCKS_DB}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/DbConfig.java]: the default value name does not match the description name.
`ROCKS_DB` vs `MAIN_DB`

File: src/main/java/com/iota/iri/conf/APIConfig.java [hunk 13]
```diff
@@ -9,43 +9,59 @@
 
 
     /**
-     * @return {@value Descriptions#PORT}
+     * Default Value: {@value BaseIotaConfig.Defaults#API_PORT}
```
Comment (alon-e) [inline src/main/java/com/iota/iri/conf/APIConfig.java]: the default value name does not match the description name.
API_PORT vs PORT
(I think I asked to change this one :) )

File: src/main/java/com/iota/iri/conf/NetworkConfig.java (hunk 10 above)
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/NetworkConfig.java]: This property will be deleted in another PR so I am not changing for now

File: src/main/java/com/iota/iri/conf/NetworkConfig.java [hunk 14]
```diff
@@ -9,68 +9,90 @@
...
     /**
-     * @return Descriptions#NEIGHBORING_SOCKET_PORT
+     * Default Value: {@value BaseIotaConfig.Defaults#NEIGHBORING_SOCKET_PORT}
+     *
+     * @return {@value Descriptions#NEIGHBORING_SOCKET_PORT}
      */
     int getNeighboringSocketPort();
 
     /**
+     * Default Value: {@value BaseIotaConfig.Defaults#RECONNECT_ATTEMPT_INTERVAL_SECONDS}
```
Comment (luca-moser) [inline src/main/java/com/iota/iri/conf/NetworkConfig.java]: @return not updated

File: src/main/java/com/iota/iri/conf/NetworkConfig.java [hunk 15]
```diff
@@ -9,68 +9,90 @@
...
      *
      * @return Descriptions#RECONNECT_ATTEMPT_INTERVAL_SECONDS
      */
     int getReconnectAttemptIntervalSeconds();
 
     /**
+     * Default Value: {@value BaseIotaConfig.Defaults#AUTO_TETHERING_ENABLED}
```
Comment (luca-moser) [inline src/main/java/com/iota/iri/conf/NetworkConfig.java]: @return not updated

File: src/main/java/com/iota/iri/conf/SnapshotConfig.java [hunk 16]
```diff
@@ -6,72 +6,99 @@
...
+     * Default Value: {@value BaseIotaConfig.Defaults#PREVIOUS_EPOCHS_SPENT_ADDRESSES_FILE}
+     *
+     * @return {@value SnapshotConfig.Descriptions#PREVIOUS_EPOCH_SPENT_ADDRESSES_FILE}
      */
     String getPreviousEpochSpentAddressesFiles();
 
     /**
+     * Default Value: {@value BaseIotaConfig.Defaults#SPENT_ADDRESSES_DB_PATH}
+     *
      * @return {@value Descriptions#SPENT_ADDRESSES_DB_PATH}
```
Comment (luca-moser) [inline src/main/java/com/iota/iri/conf/SnapshotConfig.java]: Normalize to `SnapshotConfig.Descriptions.`

File: src/main/java/com/iota/iri/conf/SnapshotConfig.java [hunk 17]
```diff
@@ -6,72 +6,99 @@
...
      */
     String getSpentAddressesDbPath();
 
     /**
+     * Default Value: {@value BaseIotaConfig.Defaults#SPENT_ADDRESSES_DB_LOG_PATH}
+     *
      * @return {@value Descriptions#SPENT_ADDRESSES_DB_LOG_PATH}
```
Comment (luca-moser) [inline src/main/java/com/iota/iri/conf/SnapshotConfig.java]: Normalize to `SnapshotConfig.Descriptions`

File: src/main/java/com/iota/iri/conf/TipSelConfig.java [hunk 18]
```diff
@@ -8,22 +8,30 @@
...
+     *
+     * @return {@value TipSelConfig.Descriptions#ALPHA}
      */
     double getAlpha();
 
     /**
-     * @return Descriptions#TIP_SELECTION_TIMEOUT_SEC
+     * Default Value: {@value BaseIotaConfig.Defaults#TIP_SELECTION_TIMEOUT_SEC}
+     *
+     * @return {@value Descriptions#TIP_SELECTION_TIMEOUT_SEC}
```
Comment (luca-moser) [inline src/main/java/com/iota/iri/conf/TipSelConfig.java]: Normalize to `TipSelConfig.Descriptions`

File: src/main/java/com/iota/iri/conf/TipSelConfig.java [hunk 19]
```diff
@@ -8,22 +8,30 @@
...
      */
     int getTipSelectionTimeoutSec();
 
     /**
-     * @return Descriptions#BELOW_MAX_DEPTH_TRANSACTION_LIMIT
+     *  Default Value: {@value BaseIotaConfig.Defaults#BELOW_MAX_DEPTH_TRANSACTION_LIMIT}
+     *
+     * @return {@value Descriptions#BELOW_MAX_DEPTH_TRANSACTION_LIMIT}
```
Comment (luca-moser) [inline src/main/java/com/iota/iri/conf/TipSelConfig.java]: Normalize to `TipSelConfig.Descriptions`

File: src/main/java/com/iota/iri/conf/NetworkConfig.java [hunk 20]
```diff
@@ -9,68 +9,90 @@
...
+     *
+     * @return {@value NetworkConfig.Descriptions#SEND_LIMIT}
      */
     int getSendLimit();
 
     /**
-     * @return Descriptions#MAX_NEIGHBORS
+     * Default Value: {@value BaseIotaConfig.Defaults#MAX_NEIGHBORS}
+     *
+     * @return {@value Descriptions#MAX_NEIGHBORS}
```
Comment (luca-moser) [inline src/main/java/com/iota/iri/conf/NetworkConfig.java]: Normalize to `NetworkConfig.Descriptions`

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java [hunk 21]
```diff
@@ -812,7 +812,7 @@ protected void setCacheSizeBytes(int cacheSizeBytes) {
 
     @Override
     public Hash getCoordinator() {
-        return Defaults.COORDINATOR_ADDRESS;
+        return Defaults.COORDINATOR;
```
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: The reason I changed it now is because of Alon's comment: https://github.com/iotaledger/iri/pull/1159#discussion_r234547241

Now I can either:
a) Disregard Alon's comment and have a discrepancy between the config name and the variable. But I think Alon was right so I'd rather not do that
b) Change the method name to be `getCoordinatorAddress` like you offered.

(b) sounds correct to me BUT:
1. This will change the configuration name in the INI file (and we also need to change the name of the flag to match). Changing the configuration names themselves shouldn't be in the context of this PR, which is just about documentation.
2. Changing config name requires both our devops and the community to change their settings and various scripts, thus inducing more work. So at the very least, an issue must be created on Github to make sure we don't miss on communicating (happened before).
By the way, if we change our configurations again, maybe we want to bundle in more changes for the same release (like https://github.com/iotaledger/iri/issues/1221) so people won't have to maintain configurations for each release. 
3. If we change the config name, note that people will have to start typing a longer name (COORDINATOR_ADDRESS, so many characters). You can argue that clarity might be more important. I would answer that I don't have really strong feelings on this last point, but I still raised it.


To summarize this long comment:
If you want this change, feel free to open an issue :-)

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java [hunk 22]
```diff
@@ -32,7 +32,7 @@
     private boolean testnet = false;
     
     //API
-    protected int port = Defaults.API_PORT;
+    protected int port = Defaults.PORT;
```
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: See coordinator comment:
https://github.com/iotaledger/iri/pull/1159#discussion_r304349780

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java [hunk 23]
```diff
@@ -969,12 +969,12 @@ protected void setPowThreads(int powThreads) {
         int POW_THREADS = 0;
 
         //Coo
-        Hash COORDINATOR_ADDRESS = HashFactory.ADDRESS.create(
+        Hash COORDINATOR = HashFactory.ADDRESS.create(
```
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: See coordinator comment:
#1159 (comment)

File: src/main/java/com/iota/iri/conf/MilestoneConfig.java [hunk 24]
```diff
@@ -9,17 +9,21 @@
 public interface MilestoneConfig extends Config {
 
     /**
-     * @return Descriptions#COORDINATOR
+     * Default Value: {@link BaseIotaConfig.Defaults#COORDINATOR}
```
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/MilestoneConfig.java]: See coordinator comment:
#1159 (comment)

File: src/main/java/com/iota/iri/conf/SnapshotConfig.java (hunk 16 above)
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/SnapshotConfig.java]: Why? Since `Descriptions` is inside `SnapshotConfig` it is clear enough which one we use I think.

I believe that in the Style Guide I wrote something against using static imports and using full names instead. This is not in the scope of what was described in the style guide.

Comment (luca-moser) [reply src/main/java/com/iota/iri/conf/SnapshotConfig.java]: Why did you use `SnapshotConfig.Descriptions` for the other getters?

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java [hunk 25]
```diff
@@ -944,7 +944,7 @@ protected void setPowThreads(int powThreads) {
         double P_PROPAGATE_REQUEST = 0.01d;
         int MWM = 14;
         int PACKET_SIZE = 1650;
-        int REQ_HASH_SIZE = 46;
+        int REQUEST_HASH_SIZE = 46;
```
[the remark this answers is not in the document]
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: I double checked with the `dev` branch and I see it is also 46.
I noticed that you now have 
`public final static int GOSSIP_REQUESTED_TX_HASH_BYTES_LENGTH = 49` hard coded and non configurable.

So I updated https://github.com/iotaledger/iri/issues/1523.

File: src/main/java/com/iota/iri/conf/SnapshotConfig.java (hunk 16 above)
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/SnapshotConfig.java]: Now you made me look and I remember why...

When I try to read the javadocs with IntelliJ there are issues if I don't use the fully qualified name
So I will change, thanks for the catch

File: src/main/java/com/iota/iri/conf/SnapshotConfig.java (hunk 17 above)
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/SnapshotConfig.java]: :+1:

File: src/main/java/com/iota/iri/conf/TipSelConfig.java (hunk 18 above)
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/TipSelConfig.java]: :+1:

File: src/main/java/com/iota/iri/conf/TipSelConfig.java (hunk 19 above)
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/TipSelConfig.java]: :+1:

File: src/main/java/com/iota/iri/conf/NetworkConfig.java (hunk 20 above)
Comment (GalRogozinski) [reply src/main/java/com/iota/iri/conf/NetworkConfig.java]: :+1:
